packages/edit-widgets/src/components/header/style.scss

```diff
@@ -103,15 +103,18 @@
 
 .edit-widgets-header-toolbar {
 	gap: $grid-unit-10;
+	// Some plugins add buttons here despite best practices.
+	// Push them a bit rightwards to fit the top toolbar.
+	margin-right: $grid-unit-10;
 
 	// The Toolbar component adds different styles to buttons, so we reset them
 	// here to the original button styles
 	// Specificity bump needed to offset https://github.com/WordPress/gutenberg/blob/8ea29cb04412c80c9adf7c1db0e816d6a0ac1232/packages/components/src/toolbar/style.scss#L76
 	> .components-button.has-icon.has-icon.has-icon,
 	> .components-dropdown > .components-button.has-icon.has-icon {
-		height: $button-size;
-		min-width: $button-size;
-		padding: 6px;
+		height: $button-size-compact;
+		min-width: $button-size-compact;
+		padding: 4px;
```

Comment:
This padding change follows this approach:

https://github.com/WordPress/gutenberg/blob/b5a31264e7c614534eb021ca272e9dca30676b70/packages/editor/src/components/document-tools/style.scss#L47